test/AutoDiff/validation-test/forward_mode.swift

```diff
@@ -1319,4 +1319,21 @@ ForwardModeTests.test("ForceUnwrapping") {
   expectEqual(5, forceUnwrap(Float(2)))
 }
 
+//===----------------------------------------------------------------------===//
+// Array methods from ArrayDifferentiation.swift
+//===----------------------------------------------------------------------===//
+ForwardModeTests.test("Array.differentiableMap") {
+  let a: Array<Float> = [1, 2, 3]
+
+  func multiplyMap(a: Array<Float>) -> Array<Float> {
+    return a.differentiableMap({x in 3 * x});
+  }
+  expectEqual([3, 3, 3], differential(at: [1, 1, 1], in: multiplyMap)(a))
+
+  func squareMap(a: Array<Float>) -> Array<Float> {
+    return a.differentiableMap({x in x * x});
+  }
+  expectEqual([2, 4, 6], differential(at: [1, 1, 1], in: squareMap)(a))
+}
+
```

Comment:
It would be great to maintain consistent Swift coding style! I recommend using [`swift-format`](https://github.com/apple/swift-format), which produces this output:

```console
$ swift-format forward_mode.swift
...

ForwardModeTests.test("Array.differentiableMap") {
  let a: [Float] = [1, 2, 3]

  func multiplyMap(a: [Float]) -> [Float] {
    return a.differentiableMap({ x in 3 * x })
  }
  expectEqual([3, 3, 3], differential(at: [1, 1, 1], in: multiplyMap)(a))

  func squareMap(a: [Float]) -> [Float] {
    return a.differentiableMap({ x in x * x })
  }
  expectEqual([2, 4, 6], differential(at: [1, 1, 1], in: squareMap)(a))
}
```

Some points, all of which are corrected by `swift-format`:
* Please use syntactic sugar `[Float]` instead of `Array<Float>`.
* Please drop semicolons.
* Please use horizontal spacing as described in [Google's Swift Style Guide](https://google.github.io/swift/#horizontal-whitespace). (Closure spacing here is corrected.)

Useful references are [Google's Swift Style Guide](https://google.github.io/swift) and the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines).